arangod/Aql/ExecutionEngine.cpp

```diff
@@ -267,14 +267,7 @@ ExecutionEngine::~ExecutionEngine() {
     std::this_thread::sleep_for(10ms);
   }
 
-  // We need to stop prefetch tasks in topological order so
-  // that after stopping tasks on a certain node there is no prefetch
-  // tasks running on any dependent which could start another on the
-  // current block.
-  // The blocks are pushed in a reversed topological order.
-  for (auto it = _blocks.rbegin(); it != _blocks.rend(); ++it) {
-    (*it)->stopAsyncTasks();
-  }
+  stopAsyncTasks();
```

Comment:
```suggestion
```

This can and should be deleted now, or am I overlooking something?

Reply:
I agree, but I think in practice this will just call the `stopAsyncTasks` and should do nothing since there should be no async task to loop through. 

But that is built under the assumption that the `ExecutionEngine` dtor will always be called after we have cleaned up the async tasks. That's why I deliberately left it, since it should be the responsibility of `ExecutionEngine` to do its own cleanup, and if somebody else did it before, that is also fine. What do you think?